rest_auth_provider.py

```diff
@@ -83,14 +82,14 @@ def check_password(self, user_id, password):
 
             # fixme: temporary fix
             try:
-                store = yield self.account_handler._hs.get_profile_handler().store  # for synapse >= 1.9.0
+                store = await self.account_handler._hs.get_profile_handler().store  # for synapse >= 1.9.0
```

Comment:
@ma1uta I am not sure about this whole `try... except` part. The async/await was introduced in Synapse v1.15.0.